compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1156,9 +1156,11 @@ public void visit(BLangSimpleVariable varNode, AnalyzerData data) {
 
         BType lhsType = varNode.symbol.type;
         varNode.setBType(lhsType);
-
         // Configurable variable type must be a subtype of anydata.
         if (configurable && varNode.typeNode != null) {
+            if (lhsType.tag == TypeTags.SEMANTIC_ERROR) {
```

Comment:
Can we have one if condition here.

ex. if (configurable && varNode.typeNode != null && lhsType.tag != TypeTags.SEMANTIC_ERROR)

Reply:
Do we need to chck varNode.typeNode != null